notebooks/how_to/how_to_run_tests/1_how_to_run_dataset_based_tests.ipynb

```diff
@@ -0,0 +1,771 @@
+       "  text-align: left;\n",
+       "}\n",
+       "</style>\n",
+       "<table id=\"T_0804a\">\n",
+       "  <thead>\n",
+       "    <tr>\n",
+       "      <th id=\"T_0804a_level0_col0\" class=\"col_heading level0 col0\" >Test Type</th>\n",
+       "      <th id=\"T_0804a_level0_col1\" class=\"col_heading level0 col1\" >Name</th>\n",
+       "      <th id=\"T_0804a_level0_col2\" class=\"col_heading level0 col2\" >Description</th>\n",
+       "      <th id=\"T_0804a_level0_col3\" class=\"col_heading level0 col3\" >ID</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <td id=\"T_0804a_row0_col0\" class=\"data row0 col0\" >Metric</td>\n",
+       "      <td id=\"T_0804a_row0_col1\" class=\"data row0 col1\" >Pearson Correlation Matrix</td>\n",
+       "      <td id=\"T_0804a_row0_col2\" class=\"data row0 col2\" >Evaluates linear dependency between numerical variables in a dataset via a Pearson Correlation coefficient heat map....</td>\n",
+       "      <td id=\"T_0804a_row0_col3\" class=\"data row0 col3\" >validmind.data_validation.PearsonCorrelationMatrix</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n"
+      ],
+      "text/plain": [
+       "<pandas.io.formats.style.Styler at 0x16f59ffd0>"
+      ]
+     },
+     "execution_count": 21,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "vm.tests.list_tests(filter=\"PearsonCorrelationMatrix\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From the output, you can see that the test ID for the pearson correlation matrix is `validmind.data_validation.PearsonCorrelationMatrix`. The `describe_test` function gives you more information about the test, including its **Required Inputs**:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "b828b4a1719b44a3a53dbf49514df171",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "HTML(value='\\n<div>\\n  <h2>Pearson Correlation Matrix</h2>\\n  <p>Evaluates linear dependency between numerical…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "test_id = \"validmind.data_validation.PearsonCorrelationMatrix\"\n",
+    "vm.tests.describe_test(test_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since this test requires a dataset, it should throw an error if you were to run it without passing a `dataset` input:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "'TestInput' object has no attribute 'dataset'\n"
```

Comment:
The message is not clear enough what I have to do here.
May be show something like this: 
"'TestInput' object has no attribute 'dataset'
Pass inputs in the following format: 
inputs={"dataset": vm_dataset} where vm_dataset is created using the vm.init_dataset() function from the ValidMind (vm) module.\n"

Reply:
Ok. Let's do that as a follow up since this error is coming directly from the `TestInput` class and we have to do a little bit of surgery to catch errors and make them more readable to the user.